packages/teams-js/test/utils.ts

```diff
@@ -138,6 +138,20 @@ export class Utils {
     return null;
   };
 
+  public findKthMessagesByFunc = (func: string, _k: number): MessageRequest | null => {
+    let k = _k;
+    for (let i = 0; i < this.messages.length; i++) {
+      if (this.messages[i].func === func) {
+        if (k === 0) {
+          return this.messages[i];
+        } else {
+          k--;
+        }
+      }
+    }
+    return null;
+  };
```

Comment:
There's nothing wrong with the way you've written this (as far as I can see anyway) but if we can prefer using the `for..of` pattern, the core of the function will more closely match canonical JS iterating patterns. Counting up rather than down could also make this function easier to understand by future readers.

Not saying you have to make the change, but consider if this might be more readable:
```
  function findKthMessagesByFunc(func: string, k: number): MessageRequest | null {
    let countOfMatchingMessages = 0;

    for (const message of messages) {
        if (message.func === func) {
            if (countOfMatchingMessages === k) {
                return message;
            }
            ++countOfMatchingMessages;
        }
    }

    return null;
  }
```

Reply:
Updated